src/main/java/gov/nasa/jpl/mbee/mdk/mms/sync/delta/DeltaSyncRunner.java

```diff
@@ -331,17 +330,22 @@ else if (shouldUpdate && jmsChange != null) {
                 }
             }
             if (postElements.size() > 0) {
-                Application.getInstance().getGUILog().log("[INFO] Queuing request to create/update " + NumberFormat.getInstance().format(postElements.size()) + " local element" + (postElements.size() != 1 ? "s" : "") + " on the MMS.");
                 URIBuilder requestUri = MMSUtils.getServiceProjectsRefsElementsUri(project);
                 try {
                     File sendData = MMSUtils.createEntityFile(this.getClass(), ContentType.APPLICATION_JSON, postElements, MMSUtils.JsonBlobType.ELEMENT_JSON);
-                    OutputQueue.getInstance().offer(new Request(project, MMSUtils.HttpRequestType.POST, requestUri, sendData, ContentType.APPLICATION_JSON, postElements.size(), "Sync Changes"));
+                    Request request = new Request(project, MMSUtils.HttpRequestType.POST, requestUri, sendData, ContentType.APPLICATION_JSON, postElements.size(), "Sync Changes");
+                    MMSUtils.sendMMSRequest(request.getProject(), request.getRequest(), progressStatus);
+//                    Application.getInstance().getGUILog().log("[INFO] Queuing request to create/update " + NumberFormat.getInstance().format(postElements.size()) + " local element" + (postElements.size() != 1 ? "s" : "") + " on the MMS.");
+//                    OutputQueue.getInstance().offer(request);
                 } catch (IOException e) {
                     Application.getInstance().getGUILog().log("[ERROR] Unexpected JSON processing exception. See logs for more information.");
                     e.printStackTrace();
                 } catch (URISyntaxException e) {
                     Application.getInstance().getGUILog().log("[ERROR] Unexpected URI syntax exception. See logs for more information.");
                     e.printStackTrace();
+                } catch (ServerException e) {
+                    Application.getInstance().getGUILog().log("[ERROR] Unexpected server exception. See logs for more information.");
+                    e.printStackTrace();
```

Comment:
Any request failure should result in the entire CSync process failing and as a result the entire changelogs persisted for next commit.